drivers: nrfwifi: Fix FW patch file linking

krish2718 opened the pull request:
Use a Zephyr native approach to link a binary file instead of a custom INCBIN approach.

nordicjm commented on drivers/wifi/nrfwifi/CMakeLists.txt:
```diff
@@ -173,15 +173,15 @@ else()
   add_custom_command(
-    OUTPUT ${CMAKE_SOURCE_DIR}/src/fw_load.c
-    COMMAND ${CMAKE_COMMAND} -E touch ${CMAKE_SOURCE_DIR}/src/fw_load.c
+    OUTPUT ${ZEPHYR_BASE}/drivers/wifi/nrfwifi/src/fw_load.c
+    COMMAND ${CMAKE_COMMAND} -E touch ${ZEPHYR_BASE}/drivers/wifi/nrfwifi/src/fw_load.c
```
In no way should this code ever overwrite files in zephyr or touch files outside of the build folder

krish2718 replied:
As discussed, reverted the original commit and switched from incbin to native approach, thanks.

krish2718 committed:
Revert "drivers: nrfwifi: Fix rebuilding when FW blobs are changes"

This reverts commit e2e96acebfef43fe619d315587592b8885a1187e.
This will be properly fixes by adding a target for nRF70.bin and
removing INCBIN approach.

Signed-off-by: Chaitanya Tata <Chaitanya.Tata@nordicsemi.no>

krish2718 committed:
drivers: nrfwifi: Use Zephyr tooling to load nRF70 FW file

Instead of relying on INCBIN macros which do not properly add
dependencies, e.g., modifying FW file doesn't trigger rebuild.

Use the Zephyr cmake tooling to load the FW patch file as a header.

This also improves memory report where the patch target is clearly
visible instead of a hidden section.

Signed-off-by: Chaitanya Tata <Chaitanya.Tata@nordicsemi.no>

nordicjm commented on drivers/wifi/nrfwifi/src/fw_load.c:
```diff
@@ -17,57 +17,19 @@
 
-	status = nrf_wifi_fmac_fw_parse(rpu_ctx, fw_start, fw_end - fw_start,
-					&fw_info);
+	status = nrf_wifi_fmac_fw_parse(rpu_ctx,
+			  nrf70_fw_patch,
+			  sizeof(nrf70_fw_patch),
+			  &fw_info);
```
nit. line length is 100 characters so multiple/all can go on one line, and if not, they should line up
